### PR TITLE
[feat, i18n] Implement ngettext

### PR DESCRIPTION
**Pull request by Frenzie**
Fixes <https://github.com/koreader/koreader/issues/5249>.

Looks a little something like this:
Screenshot_2019-08-24_18-58-11
Screenshot_2019-08-24_18-58-18

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -466,7 +466,8 @@ XGETTEXT_BIN=xgettext
 
 pot:
 	mkdir -p $(TEMPLATE_DIR)
-	$(XGETTEXT_BIN) --from-code=utf-8 --keyword=C_:1c,2 \
+	$(XGETTEXT_BIN) --from-code=utf-8 \
+		--keyword=C_:1c,2 --keyword=N_:1,2 \
```
Comment (Frenzie) [inline Makefile]: For `npgettext`, given that we're using `C_` for `pgettext`, logically that should be `NC_`. I didn't really test that one yet but other than the lacking npgettext method it should be working.

File: frontend/gettext.lua [hunk 2]
```diff
@@ -57,20 +145,43 @@ function GetText_mt.__index.changeLang(new_lang)
...
-                else
-                    GetText.translation[data.msgid] = unescaped_string
                 end
+            elseif data.msgid and data.msgstr and data.msgstr ~= "" then
+                -- header
+                if not headers and data.msgid == "" then
+                    local util = require("util")
+                    headers = data.msgstr
+                    local plural_forms = data.msgstr:match("Plural%-Forms: (.*);")
+                    local nplurals = plural_forms:match("nplurals=([0-9]+);") or 1
```
Comment (Frenzie) [inline frontend/gettext.lua]: 2 actually seems to be right; I'll have to look at the docs a bit better. Clearly you don't have to declare the default.

Comment (Frenzie) [reply frontend/gettext.lua]: Oh, I guess perhaps I can't check for sanity because a C boolean expression is 0 or 1.

File: Makefile (hunk 1 above)
Comment (poire-z) [reply Makefile]: But `CN_` would state the order: first context, then the next thing will be N'ed :)
(if we don't need to care about the involved gettext function names - but you decide)

Comment (Frenzie) [reply Makefile]: For reference, GTK also has an NC_ (although it means something completely different): https://developer.gnome.org/glib/stable/glib-I18N.html#NC-:CAPS

Anyway, I don't even know if we need `npgettext`. It's just with all the groundwork already there it seems silly not to add the method.

File: plugins/coverbrowser.koplugin/listmenu.lua [hunk 3]
```diff
@@ -381,7 +382,7 @@ function ListMenuItem:update()
                 end
             else
                 if pages then
-                    pages_str = T(_("%1 pages"), pages)
+                    pages_str = T(N_("1 page", "%1 pages", pages), pages)
```
Comment (Frenzie) [inline plugins/coverbrowser.koplugin/listmenu.lua]: @poire-z Like so:

Screenshot_2019-08-24_23-00-42


Screenshot_2019-08-24_22-58-43

**Comment by Frenzie** [comment]
Merging now so the translations can start rolling in; code should work as designed per-spec but we'll see if anything funny crops up in the wild. ;-)

**Comment by Frenzie** [comment]
And it looks like this on Transifex:

Screenshot_2019-08-24_23-21-21

**Pull request merged**